feign-opentracing/src/test/java/feign/opentracing/AbstractFeignTracingTest.java

```diff
@@ -0,0 +1,166 @@
+
+        List<MockSpan> mockSpans = mockTracer.finishedSpans();
+        Assert.assertEquals(1, mockSpans.size());
+
+        RecordedRequest recordedRequest = mockWebServer.takeRequest();
+        Assert.assertEquals(mockSpans.get(0).context().spanId(),
+                Long.parseLong(recordedRequest.getHeader("spanId")));
+        Assert.assertEquals(mockSpans.get(0).context().traceId(),
+                Long.parseLong(recordedRequest.getHeader("traceId")));
+    }
+
+    @Test
+    public void testParentSpanFromSpanManager() throws InterruptedException {
+        {
+            MockSpan parent = mockTracer.buildSpan("parent").start();
+            DefaultSpanManager.getInstance().activate(parent);
+
+            mockWebServer.enqueue(new MockResponse()
+                    .setResponseCode(200));
+
+            StringEntityRequest
+                    entity = feign.<StringEntityRequest>newInstance(new Target.HardCodedTarget(StringEntityRequest.class,
+                    mockWebServer.url("/foo").toString()));
+            entity.get();
+            parent.finish();
+        }
+
+        List<MockSpan> mockSpans = mockTracer.finishedSpans();
+        Assert.assertEquals(2, mockSpans.size());
+        Assert.assertEquals(mockSpans.get(1).context().traceId(), mockSpans.get(0).context().traceId());
+        Assert.assertEquals(mockSpans.get(1).context().spanId(), mockSpans.get(0).parentId());
+    }
+
+    @Test
+    public void testUnknownHostException() {
+        {
+            StringEntityRequest entity =
+                    feign.<StringEntityRequest>newInstance(new Target.HardCodedTarget(StringEntityRequest.class,
+                    "http://www.abcfoobar.bar/baz"));
+            try {
+                entity.get();
+            } catch (Exception ex) {
+                //ok
+            }
+        }
+
+        List<MockSpan> mockSpans = mockTracer.finishedSpans();
+        // there are two spans due to retry mechanism
+        Assert.assertEquals(2, mockSpans.size());
+        assertErrorSpan(mockSpans.get(0), "http://www.abcfoobar.bar/baz");
+        assertErrorSpan(mockSpans.get(1), "http://www.abcfoobar.bar/baz");
+    }
+
+    public static void assertErrorSpan(MockSpan mockSpan, String url) {
+        Assert.assertEquals(5, mockSpan.tags().size());
+        Assert.assertNotNull(mockSpan.tags().get(Tags.COMPONENT.getKey()));
+        Assert.assertEquals(Tags.SPAN_KIND_CLIENT, mockSpan.tags().get(Tags.SPAN_KIND.getKey()));
+        Assert.assertEquals("GET", mockSpan.tags().get(Tags.HTTP_METHOD.getKey()));
+        Assert.assertEquals(url, mockSpan.tags().get(Tags.HTTP_URL.getKey()));
+        Assert.assertEquals(Boolean.TRUE, mockSpan.tags().get(Tags.ERROR.getKey()));
+
+        Assert.assertEquals(1, mockSpan.logEntries().size());
+        Assert.assertEquals(Tags.ERROR.getKey(), mockSpan.logEntries().get(0).fields().get("event"));
+        Assert.assertNotNull(mockSpan.logEntries().get(0).fields().get("error.object"));
```

Comment:
Agree, it is from here: https://github.com/opentracing/specification/blob/master/semantic_conventions.md#captured-errors